### PR TITLE
Use volk and #pragma once

### PR DESCRIPTION
**Pull request by charles-lunarg**
test: Use volk instead of custom function loader

Volk is a very common library for managing loading of vulkan and its
associated function pointers.
Using Volk reduces the changes occuring from header updates.

code: Use #pragma once everywhere

Inconsistently used #pragma once in some files, and use header guards in
others. Best to make everything consistent, and use #pragma once
everywhere.

### CODE REVIEW

File: tests/extension_layer_tests.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ bool ImageFormatAndFeaturesSupported(const VkInstance inst, const VkPhysicalDevi
     // Verify again using version 2, if supported, which *can* return more property data than the original...
     // (It's not clear that this is any more definitive than using the original version - but no harm)
     PFN_vkGetPhysicalDeviceImageFormatProperties2KHR p_GetPDIFP2KHR =
-        (PFN_vkGetPhysicalDeviceImageFormatProperties2KHR)vk::GetInstanceProcAddr(inst,
+        (PFN_vkGetPhysicalDeviceImageFormatProperties2KHR)vkGetInstanceProcAddr(inst,
```
Comment (jeremyg-lunarg) [inline tests/extension_layer_tests.cpp:114]: Can't volk get rid of the need for this vkGetInstanceProcAddr call?

File: tests/vkrenderframework.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ void DebugReporter::Create(VkInstance instance) noexcept {
     assert(instance);
     assert(!debug_obj_);
 
-    auto DebugCreate = reinterpret_cast<DebugCreateFnType>(vk::GetInstanceProcAddr(instance, debug_create_fn_name_));
+    auto DebugCreate = reinterpret_cast<DebugCreateFnType>(vkGetInstanceProcAddr(instance, debug_create_fn_name_));
     if (!DebugCreate) return;
```
Comment (jeremyg-lunarg) [inline tests/vkrenderframework.cpp:241]: another vkGetInstanceProcAddr that seems llike it could go away.

File: tests/vkrenderframework.cpp [hunk 3]
```diff
@@ -248,7 +248,7 @@ void DebugReporter::Destroy(VkInstance instance) noexcept {
     assert(instance);
     assert(debug_obj_);  // valid to call with null object, but probably bug
 
-    auto DebugDestroy = reinterpret_cast<DebugDestroyFnType>(vk::GetInstanceProcAddr(instance, debug_destroy_fn_name_));
+    auto DebugDestroy = reinterpret_cast<DebugDestroyFnType>(vkGetInstanceProcAddr(instance, debug_destroy_fn_name_));
```
Comment (jeremyg-lunarg) [inline tests/vkrenderframework.cpp:251]: another vkGetInstanceProcAddr that seems llike it could go away.

File: tests/vktestbinding.h [hunk 4]
```diff
@@ -207,7 +206,7 @@ class Device : public internal::Handle<VkDevice> {
     bool IsEnabledExtension(const char *extension);
 
     // vkGetDeviceProcAddr()
-    PFN_vkVoidFunction get_proc(const char *name) const { return vk::GetDeviceProcAddr(handle(), name); }
+    PFN_vkVoidFunction get_proc(const char *name) const { return vkGetDeviceProcAddr(handle(), name); }
```
Comment (jeremyg-lunarg) [inline tests/vktestbinding.h:209]: This method is unused. Should it be removed?

**Review by jeremyg-lunarg — COMMENTED** [review]
This is great! yay volk!

There's a few remaining GetProcAddr calls that seem like the could be removed. If they can't, please add comments to the code explaining why.

**Comment by charles-lunarg** [comment]
Looks like I missed a few - happy to remove those. First have to figure out why android doesn't like me.

File: tests/vktestframework.cpp [hunk 5]
```diff
@@ -102,7 +102,7 @@ int fopen_s(FILE **pFile, const char *filename, const char *mode) {
 void TestEnvironment::SetUp() {
     // Initialize GLSL to SPV compiler utility
     glslang::InitializeProcess();
-    vk::InitDispatchTable();
+    volkInitialize();
```
Comment (juan-lunarg) [inline tests/vktestframework.cpp:105]: Check return value of volkInitialize

**Comment by juan-lunarg** [comment]
> Looks like I missed a few - happy to remove those. First have to figure out why android doesn't like me.

Running Android locally and I can't repro the failure on my Pixel 3

**Comment by juan-lunarg** [comment]
Here is something weird though:

```
INFO: executing #1: cmake -H. -B/home/lunarg/.jenkins/wvel/Debug64/Vulkan-ExtensionLayer/android-build -DCMAKE_INSTALL_PREFIX=/home/lunarg/.jenkins/wvel/Debug64/Vulkan-ExtensionLayer/android-build/install -DBUILD_TESTS=ON -DUPDATE_DEPS=ON -DBUILD_TESTS=ON -DCMAKE_TOOLCHAIN_FILE=/home/lunarg/Android/Sdk/ndk/25.2.9519653/build/cmake/android.toolchain.cmake -DANDROID_USE_LEGACY_TOOLCHAIN_FILE=NO -DANDROID_PLATFORM=26 -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a -DCMAKE_ANDROID_STL_TYPE=c++_static -DCMAKE_INSTALL_LIBDIR=lib/arm64-v8a -DCMAKE_BUILD_TYPE=Release
```

It's using `Release` instead of `Debug` which makes figuring out what's going on more difficult

**Review by jeremyg-lunarg — APPROVED** [review]

**Comment by juan-lunarg** [comment]
Note we also fixed Jenkins to use Debug instead of Release

Thank you @lunarpapillo 👍🏾

**Review by juan-lunarg — APPROVED** [review]
🥇

**Pull request merged**